components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -909,7 +902,29 @@ private Condition buildSearchCondition(SearchCondition searchCondition) {
         if (!(searchCondition.getStatement() == null)) {
             PrimitiveStatement primitiveStatement = searchCondition.getStatement();
 
-            if (Constants.SEARCH_KEYS.contains(primitiveStatement.getProperty())) {
+            if (Constants.SEARCH_KEY_NAME.equals(primitiveStatement.getProperty())) {
+                return new PrimitiveCondition(
+                        Constants.SEARCH_KEY_NAME_INTERNAL, getPrimitiveOperatorFromOdata(primitiveStatement
+                        .getCondition()), primitiveStatement.getValue());
+            } else if (Constants.SEARCH_KEY_SERVICES.equals(primitiveStatement.getProperty()) &&
+                    getPrimitiveOperatorFromOdata(primitiveStatement.getCondition()) == EQUALS) {
+                Condition servicesAttrKeyCondition = new PrimitiveCondition(Constants.ATTR_KEY, EQUALS, Constants
+                        .SEARCH_KEY_SERVICES);
+                Condition servicesAttrValueCondition = new PrimitiveCondition(Constants.ATTR_VALUE, EQUALS,
+                        primitiveStatement.getValue());
+                Condition servicesCondition = new ComplexCondition(getComplexOperatorFromOdata(ConditionType.AND),
+                        Arrays.asList(servicesAttrKeyCondition, servicesAttrValueCondition));
+
+                Condition servicesCombinedAttrKeyCondition = new PrimitiveCondition(Constants.ATTR_KEY, EQUALS,
+                        Constants.SEARCH_KEY_SERVICES);
+                Condition servicesCombinedAttrValueCondition = new PrimitiveCondition(Constants.ATTR_VALUE, EQUALS,
+                        SERV_PROVISIONING + "," + SERV_AUTHENTICATION);
```

Comment:
Can we have this combination as a constant... so when adding and searching we have the same order.

Reply:
added a constant.. but however when adding, we are adding them to a list and calling String.join.. 
https://github.com/wso2/identity-api-server/blob/master/components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java#L1182